src/main/java/net/minecraftforge/event/entity/MovementCheckEvent.java

```diff
@@ -0,0 +1,35 @@
+package net.minecraftforge.event.entity;
+
+import net.minecraft.entity.Entity;
+import net.minecraftforge.fml.common.eventhandler.Cancelable;
+
+/**
+ * This event is fired when the server does a movement check. There are two
+ * movement checks, one for players and one for vehicles. The type can be
+ * determined using {@link #isVehicleCheck()}. If canceled, the check will be
+ * skipped and no corrective action will be applied.
+ */
+@Cancelable
+public class MovementCheckEvent extends EntityEvent 
+{
+	/**
+	 * Whether or not the check is for a vehicle.
+	 */
+	private final boolean vehicle;
```

Comment:
protected?

Reply:
What benefit would moving to protected have here? There isn't really a point to subclassing this event. If new movement checks are added in the future, the field should switch from a boolean to an enum.

Reply:
Can you just add an enum? That way mods can use enumhelper to add new types.